### DISCUSSION TITLE
What is the best way to access a sub group for type hints?

### DISCUSSION THREAD

**Discussion by haarisr** [zarr-developers/zarr-python#2055] [Q&A]
```python3   
 def write(root: zarr.Group, data:np.ndarray):
        group = root["foo/bar"]
        group.array("baz", data)
```

I have already create `root["foo/bar"]`. When I want to access it, the type hint says that `group` is either an `Array` or `Group`. How can I just extract a group?

**Comment by d-v-b** [comment]
the type checker cannot know that `root["foo/bar"]` will be a zarr group. Consider the possibility that some other program on your computer deleted the group at `foo/bar`, or replaced it with an array. But if you are sure that it's a group, you can use [`typing.cast`](https://docs.python.org/3/library/typing.html#typing.cast) to assert this to the type checker.

In the future, we could add a `get_group(group_name)` method that either returns a group or errors. this would be type-safe.
